lib/plek.rb

```diff
@@ -159,6 +162,24 @@ def find_uri(*args)
 
 private
 
+  # TODO: clean up call sites throughout alphagov and then delete clean_name.
+  def clean_name(service)
+    service.to_s.downcase.strip.gsub(/[^a-z.-]+/, "")
+  end
+
+  def should_prefix(name)
+    !@unprefixable_hosts.include?(name)
+  end
+
+  def should_use_http_for_domain(domain)
+    normalised_domain = domain.delete_suffix(".")
```

Comment:
How come we need this? I'd guess we've got a problem that the URLs don't work if they have a trailing period?